### PR TITLE
feature(editor): set route to in progress on edit, warn about unapproved routes when exporting

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing
- [x] The description lists all relevant PRs included in this release _(remove this if not merging to master)_

### Description

Two main changes:
- When any part of a route gets edited, the progress gets set to In Progress.
- In the Create Snapshot window, any in progress or pending routes will be displayed as a warning, since they will not be included in the published result. 

The snapshot pane directly requests the list of routes from the backend, which requires a lock. Therefore, in the Editor Feed Source Panel, a lock is requested before opening the Snapshot pane. After closing, the lock is removed.

### CODE REVIEW

**Comment by daniel-heppner-ibigroup** [comment]
A few issues: Why are the tests failing? It looks like it is not related to my code.
Also, I'm having a type issue on line 78 in CreateSnapshotModal.js that shows up in VSCode. 
"Cannot call `fetchRouteEntities().then` because property `then` is missing in  function [1].Flow(InferError)"

File: lib/editor/components/CreateSnapshotModal.js
Comment (miles-grant-ibigroup) [inline lib/editor/components/CreateSnapshotModal.js:78]: The tests are failing because the flow checks are failing because types here are wrong! The method doesn't return the data (since it does dispatch), which results in a race condition! 

A new dispatch action that moves handling all this state to redux is probably the cleanest way to fix it, but I will leave it up to second reviewer to decide the best way to handle this (could also try to refactor `fetchRouteEntites` to return correct type?)

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
This is looking good! And is a good feature. Unit tests are failing because the flow checks are failing. Fixing them might be a bit of a struggle. I have 2 ideas, but I think I'll leave it up to you/the second reviewer to decide which option is best.

Otherwise, all other comments are mostly nit-picks. Good work overall!

File: lib/editor/reducers/data.js
Comment (miles-grant-ibigroup) [reply lib/editor/reducers/data.js:217]: Also, what's `1`?

File: lib/editor/actions/active.js
Comment (daniel-heppner-ibigroup) [reply lib/editor/actions/active.js:233]: I did just copy this from somewhere else, but I deleted a bunch of stuff so it's not really following the pattern anymore. I think we can take the liberty of cleaning it up a bit more.

File: lib/editor/components/CreateSnapshotModal.js
Comment (daniel-heppner-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js:14]: I just copied this from somewhere, I think, but I agree.

File: lib/editor/components/CreateSnapshotModal.js
Comment (daniel-heppner-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js:81]: There is data that is stored in YAML file about GTFS spec. I can try to import it from there.

File: lib/editor/components/CreateSnapshotModal.js
Comment (daniel-heppner-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js:163]: Not just somtimes... the onChange fucntion seems to be completely missing. I have no idea how that disappeared!

File: lib/editor/components/CreateSnapshotModal.js
Comment (daniel-heppner-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js:14]: Oh, now I see, the reason we do this is because we have other variables called createSnapshot, so it's better to have this in an object.

File: lib/editor/reducers/data.js
Comment (daniel-heppner-ibigroup) [reply lib/editor/reducers/data.js:217]: 1 is Pending Approval

File: lib/editor/components/CreateSnapshotModal.js
Comment (daniel-heppner-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js:14]: Resolve if you agree

Comment (miles-grant-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js:14]: good call! that's probably cleaner.

File: lib/editor/components/CreateSnapshotModal.js
Comment (miles-grant-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js:81]: oh good idea! there's a method in a utils file as well to easily get access to that info.

File: lib/editor/reducers/data.js
Comment (miles-grant-ibigroup) [reply lib/editor/reducers/data.js:217]: Yeah I think it's' best if we pull the numbers from the spec file

**Comment by daniel-heppner-ibigroup** [comment]
I think we finally got there in the end!

File: lib/common/util/config.js
Comment (miles-grant-ibigroup) [inline lib/common/util/config.js:101]: Same issue here as above -- I'm weary that this PR will add the polish translation when that should've been done in #735

File: lib/editor/actions/active.js
Comment (miles-grant-ibigroup) [inline lib/editor/actions/active.js:211]: We already have the state object from above. Could we just re-use that object (on the line below as well)?

File: lib/editor/components/CreateSnapshotModal.js
Comment (miles-grant-ibigroup) [inline lib/editor/components/CreateSnapshotModal.js:76]: We should probably add a comment here about why we're doing this!

File: lib/editor/components/CreateSnapshotModal.js
Comment (miles-grant-ibigroup) [inline lib/editor/components/CreateSnapshotModal.js:107]: Same comment as earlier -- we should probably move `routeStatusTextToValue` to return some sort of dictionary.

File: lib/editor/components/CreateSnapshotModal.js
Comment (miles-grant-ibigroup) [inline lib/editor/components/CreateSnapshotModal.js:148]: This looks so cool!

File: lib/editor/components/CreateSnapshotModal.js
Comment (miles-grant-ibigroup) [inline lib/editor/components/CreateSnapshotModal.js:156]: Glad we moved to `getEntityName`, but I think we should include the route longName if possible! I think there's a helper method somewhere. As it stands the following can happen (if there's 2 routes with the same shortName)
Screen Shot 2021-12-06 at 2 10 04 PM

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Thank you for all those changes! Things are feeling much better. Just a few small clean-up nitpicks.

File: lib/editor/components/CreateSnapshotModal.js
Comment (daniel-heppner-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js:156]: getEntityName *should* use the routeLongName... but it doesn't use routeDesc, which some agencies use in lieu of the long name. 

https://github.com/ibi-group/datatools-ui/issues/745

Comment (daniel-heppner-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js:156]: In your screenshot, do those routes have long names or descriptions? If they are long names, they should be shown.

Comment (daniel-heppner-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js]: I think this should be fixed in a separate PR.

Comment (miles-grant-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js]: I agree! Thanks for opening #745. Will resolve this for now.

File: lib/editor/util/index.js
Comment (miles-grant-ibigroup) [inline lib/editor/util/index.js:158]:
```suggestion
export function getRouteStatusDict (): { [string]: string } {
```
Does this work as a better type than `any`?

**Review by miles-grant-ibigroup — APPROVED** [review]
Thanks for the changes! This is looking really strong. 

I just have one last comment to avoid an `any` type, but that's not blocking so feel free to assign to the second reviewer once you've made that change!

File: lib/editor/util/index.js
Comment (daniel-heppner-ibigroup) [reply lib/editor/util/index.js]: Flow seems happy with it! :)

File: lib/editor/components/CreateSnapshotModal.js
Comment (binh-dam-ibigroup) [inline lib/editor/components/CreateSnapshotModal.js:165]: Sort props.

File: lib/editor/components/FeedInfoPanel.js
Comment (binh-dam-ibigroup) [inline lib/editor/components/FeedInfoPanel.js:21]: This should probably be below `import {entity}...`

File: lib/editor/reducers/data.js
Comment (binh-dam-ibigroup) [inline lib/editor/reducers/data.js:15]: Move this to between lines 9 and 10.

File: lib/editor/reducers/data.js
Comment (binh-dam-ibigroup) [inline lib/editor/reducers/data.js:221]: Maybe rename to `additionalProps`?

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Looks mostly good, see various comments, the blocking one being about moving `onClose` to the component state. Also is the argument syntax for the modal `open` function correct?

File: lib/editor/components/CreateSnapshotModal.js
Comment (daniel-heppner-ibigroup) [reply lib/editor/components/CreateSnapshotModal.js:85]: onClose isn't even needed anymore, removing it.

File: lib/editor/components/CreateSnapshotModal.js
Comment (binh-dam-ibigroup) [inline lib/editor/components/CreateSnapshotModal.js]: import from actions, common, then util.

**Review by binh-dam-ibigroup — APPROVED** [review]
Approving subject to fixing one more import order.

**Pull request merged**
